lib/services/device/device-install-app-service.ts

```diff
@@ -49,12 +48,38 @@ export class DeviceInstallAppService {
 			device,
 			projectDir: projectData.projectDir,
 		});
+		const buildOutputOptions = platformData.getValidBuildOutputData(buildData);
+		const outputPath = buildData.outputPath || platformData.getBuildOutputPath(buildData);
+		const packages = await this.$buildArtifactsService.getAllAppPackages(
+			outputPath,
+			buildOutputOptions
+		);
+		let packageFile;
+		if (packages.length === 1) {
+			// will always be the case on iOS
+			packageFile = packages[0].packageName;
+		} else if (device.deviceInfo.abis) {
+			device.deviceInfo.abis.every(abi=>{
+				const index = packages.findIndex(p => p.packageName.indexOf(abi) !== -1);
+				if (index !== -1) {
+					packageFile = packages[index].packageName;
+					return false;
+				}
+				return true;
+			})
```

Comment:
This feels confusing to read, can we do something more readable here?
```ts
packages.find(({ packageName })=> {
  if(device.deviceInfo.abis.some(abi => packageName.includes(abi)) {
   packageFile = packageName;
   return true;
  }
})
```

(note: untested, just felt this may be easier to read)

Reply:
i am ok with the solution. The only difference i see is if the order of packages is not the same as abis.
In the original we get the first package matching the "first" abi
In the second we get the first package which is part of the abis